pkg/manager/impl/execution_manager.go

```diff
@@ -234,26 +234,36 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	}
 	taskResourceLimits := runtimeInterfaces.TaskResourceSet{}
 
-	// For resource values, we prefer to use the limits set in the application config over the set resource values.
-	if !configResourceLimits.CPU.IsZero() {
+	// For resource values, we prefer to set unset limits to the request values over the config values.
+	if cpuIndex >= 0 {
+		cpuLimit = resourceRequestEntries[cpuIndex].Value
+		cpu, err := resource.ParseQuantity(cpuLimit)
+		if err != nil {
+			logger.Errorf(ctx, "Failed to parse user cpu limit from task spec [%s] with err [%+v]", cpuLimit, err)
```

Comment:
```suggestion
			logger.Errorf(ctx, "In Task [%s] Failed to parse specified CPU limits [%s]. Resources should be valid Kubernetes quantities. Reason [%s]", task.Name, cpuLimit, err.String())
```

maybe `task.Name` is incorrect, but please provide the actual name of the task. Errors are extremely important. Obtuse errors just increase confusion

Reply:
updated